### PR TITLE
[Web] OSK key layer assignment fix

### LINKED ISSUE #1628: [Developer] version 11 gives wrong output for "Test Keyboard on web"
**Describe the bug**
When using "Test Keyboard on web", version 10 works correctly, but version 11 gives wrong output

**To Reproduce**
With Keyman Developer 10.0.1206.0 installed, loaded the rac_pashai keyboard (release/rac/rac_pashai, originally from keyboard PR #618, now merged to master).
For .kmn file, Build, Compile Keyboard, Test Keyboard on web.
Using Chrome (version 63.0.3239.73) on an iphone 4s (with iOS 9.3.5), I navigated to the address given by Developer (192.168.1.10:8008).
Keys worked as expected, particularly the top two rows of the 'symbol' layer.
Repeated using Chrome (72.0.3626.105) on a Lenovo A10 tablet (Android 6.0).
Keys worked as expected, particularly the top two rows of the 'symbol' layer.

Uninstalled Keyman Developer 10.0.1206.0 and installed 11.0.1350.0.
With Keyman Developer 11.0.1350.0 installed, loaded the rac_pashai keyboard.
For .kmn file, Build, Compile Keyboard, Test Keyboard on web.
Using Chrome (version 63.0.3239.73) on an iphone 4s (with iOS 9.3.5), I navigated to the address given by Developer (192.168.1.10:8008).
Keys worked as expect, except for the top two rows of the 'symbol' layer:
In the top row, keys labeled 1234567890 produced Arabic style digits: U+06F1..U+06F9 and U+06F0
In the second row, keys labeled @#%&* produced Arabic style digits: U+06F2, U+06F3, U+06F5, U+06F7, U+06F8
Repeat using Chrome (72.0.3626.105) on a Lenovo A10 tablet (Android 6.0).
Same problems with the top two rows of the symbol layer.

When I installed Keyman for iOS on the iphone 4s (build 10.0.211), and installed the keyboard, the keys worked as expected.

**Expected behavior**
Using the touch keys labeled 123... @#%&* should produce what is on the key, which works when using Developer version 10, but not with 11.0.1350.0.

---

**iPhone/iPad/Android (please complete the following information):**
 - Device: iPhone4s
 - OS: iOS 9.3.5
 - Target application: Chrome (version 63.0.3239.73)
 - Device: Lenovo A10 tablet
 - OS: Android 6.0
 - Target application: Chrome (version 72.0.3626.105)

**Keyman Developer (please complete the following information):
 - OS: Windows 10 Pro 1803 build 17134.590
 - Keyman Developer version: 10.0.1206.0 and 11.0.1350.0
 
---
 
**Keyboard**
 - Keyboard name: rac_pashai
 - Keyboard version: 1.0
 - Language name: Pashai

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #1628 for 11.0's stable version.

### CODE REVIEW

**Comment by keyman-server** [comment]
[OSK renders for Windows](https://build.palaso.org/repository/download/KeymanDesktop_TestPrRenderOnScreenKeyboards/142586:id/kvk/index.html)

**Comment by jahorton** [comment]
Reminder:  11.0 stable still has those macOS-Safari unit test checks in place and enabled that are currently falling over with BrowserStack.  No errors were reported for any tests that actually ran.

File: web/source/kmwosk.ts [hunk 1]
```diff
@@ -213,7 +214,7 @@ namespace com.keyman {
 
     getId(): string {
       // Define each key element id by layer id and key id (duplicate possible for SHIFT - does it matter?)
-      return this.spec.layer+'-'+this.spec.id;
+      return this.layer+'-'+this.spec.id;
```
Comment (darcywong00) [inline web/source/kmwosk.ts]: Is it possible for `this.layer` to be undefined?

Can the constructor in l. 31 initialize?
```
this.layer = 'default';
```

**Comment by jahorton** [comment]
I made sure to visually compare and contrast against the relevant 10.0 code (thanks, stable-10.0) to ensure that it better mirrors its original, pre-refactored state.  I discovered that when making the diacritic display fix, I erroneously merged two versions of 'layer' info into one field - this change directly reverts exactly that error and nothing else.

For direct testing:  I've overwritten Developer's version of KMW with a freshly-compiled copy from this PR and reloaded the testing page, after which the reported error is resolved.  (Took a bit of hunting, but I found the correct path and folder for it.)  I've also made sure to run KMW's full unit-test suite against the changes.  Finally, I made sure it didn't break other keyboards through direct web-based testing - EuroLatin was probably the main test case I used due to concern for popup keys, but there was a smattering of the other keyboards from the testing page.

The unit-tests themselves brought out extra issues for 'master', as all the extra OSK refactoring meant there were a few extra details to track.  Those have been resolved within their PR.  It probably wouldn't hurt to add a new unit-test set (for touch platforms) with similar layer info to the 'problem' keyboard, though.  (The issue's trigger:  the affected keys were on the 'symbols' layer while having custom-set 'shift' / 'right-alt' layer properties; KMW overwrote the latter with the former.)

**Comment by mcdurdin** [comment]
Thanks @jahorton, that's excellent detail. In hindsight, the overloaded layer option for keys has proven tricky to keep properly delineated for both keyboard developers and Keyman team developers ... :(

**Pull request merged**